sql/src/test/java/org/apache/druid/quidem/DruidQuidemCommandHandler.java

```diff
@@ -245,6 +269,159 @@ static class DruidPlanCommand extends AbstractRelPlanCommand
     }
   }
 
+  static class HintPlanCommand extends AbstractRelPlanCommand
+  {
+    HintPlanCommand(List<String> lines, List<String> content)
+    {
+      super(lines, content, DruidHook.DRUID_PLAN);
+    }
+
+    @Override
+    protected String getString(RelNode node)
+    {
+      final List<String> hintsCollect = new ArrayList<>();
+      final HintCollector collector = new HintCollector(hintsCollect);
+      node.accept(collector);
+      StringBuilder builder = new StringBuilder();
+      for (String hintLine : hintsCollect) {
+        builder.append(hintLine).append("\n");
+      }
+
+      return builder.toString();
+    }
+
+    private static class HintCollector extends RelShuttleImpl
+    {
+      private final List<String> hintsCollect;
+
+      HintCollector(List<String> hintsCollect)
+      {
+        this.hintsCollect = hintsCollect;
+      }
+
+      @Override
+      public RelNode visit(TableScan scan)
+      {
+        if (!scan.getHints().isEmpty()) {
+          this.hintsCollect.add("TableScan:" + scan.getHints());
+        }
```

Comment:
note: you could remove all these conditionals; add a `collectHints` method with kinda the same content; and use `relNode.getClass().getSimpleName()` to get `TableScan` and similar names

Reply:
This would not give the abstracted class names, Initially I copied this calcite. But i guess for our case we doent need to have all those implementations. removed and modified as mentioned in the other comments